### PR TITLE
Template query to not limit by days

### PR DESCRIPTION
**Pull request by minglis**
Fix for:

https://www.pivotaltracker.com/story/show/146615775

Note needs further work in:

https://www.pivotaltracker.com/story/show/146850973 to have a longer term solution.

### CODE REVIEW

File: app/dao/notifications_dao.py [hunk 1]
```diff
@@ -103,13 +103,26 @@ def create_notification_statistics_dict(service_id, day):
...
     table = NotificationHistory
 
-    if limit_days and limit_days <= 7:  # can get this data from notifications table
+    if limit_days is not None and limit_days <= 7:
         table = Notification
 
-    query_filter = [table.service_id == service_id, table.key_type != KEY_TYPE_TEST]
-    if limit_days is not None:
+        # only limit days if it's not seven days, as 7 days == the whole of Notifications table.
+        if limit_days != 7:
```
Comment (servingUpAces) [inline app/dao/notifications_dao.py]: I think the if statement on line 125 will do this, we could get rid of this one.

File: app/dao/notifications_dao.py [hunk 2]
```diff
@@ -103,13 +103,26 @@ def create_notification_statistics_dict(service_id, day):
...
+
+    elif limit_days is not None:
+        # case where not under 7 days, so using NotificationsHistory so limit allowed
+        query_filter.append(table.created_at >= days_ago(limit_days))
+
+    query_filter.append(table.service_id == service_id)
+    query_filter.append(table.key_type != KEY_TYPE_TEST)
+
+    # only limit days if it's not seven days, as 7 days == the whole of Notifications table.
+    if limit_days is not None and limit_days != 7:
```
Comment (servingUpAces) [inline app/dao/notifications_dao.py]: Do we need tests in test_notification_dao to test limit_days=7?

File: app/dao/notifications_dao.py (hunk 1 above)
Comment (minglis) [inline app/dao/notifications_dao.py]: it was reading really weird so I overwrote the code.

File: app/dao/notifications_dao.py (hunk 2 above)
Comment (minglis) [inline app/dao/notifications_dao.py]: think exitsing tests have this case

**Pull request merged**